resolvers.js

```diff
@@ -526,6 +535,37 @@ const Resolvers = {
         logger.error(e.message);
         return e.message;
       }
+    },
+
+    async deleteOpticalImage(_, args) {
+      let {datasetId} = args;
+      const payload = jwt.decode(args.jwt, config.jwt.secret);
+      const basePath = `http://localhost:${config.img_storage_port}`;
+      try {
+        await checkPermissions(datasetId, payload);
+        let opticalImageIds = await pg.select('id').from('optical_image').where('ds_id', '=', datasetId);
+        let rawOpticalImage = await pg.select('optical_image').from('dataset').where('id','=',datasetId);
+        Array.prototype.push.apply(opticalImageIds, rawOpticalImage);
+        opticalImageIds.forEach((Image) => {
+          if (Image.id){ //to delete zoomed Image
+            let url = basePath + `${config.img_upload.categories.optical_image.path}delete/${Image.id}`;
+            fetch(url, {
```

Comment:
@Ren22 Try to await for the result of "fetch" to ensure that exceptions inside a promise are handled well.
And call "fetch" on an incorrect url to see if an exception is raised.